### PR TITLE
Hyper bump

### PR DESCRIPTION
**Pull request by avadacatavra**
bump hyper to 0.10

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 
 name = "websocket"
-version = "0.17.1"
+version = "0.18.0"
```
Comment (nox) [inline Cargo.toml:4]: Rebase, current version is already `0.18.0`.

File: src/client/mod.rs [hunk 2]
```diff
@@ -86,7 +86,8 @@ impl Client<DataFrame, Sender<WebSocketStream>, Receiver<WebSocketStream>> {
 		));
 
 		let stream = if secure {
-			let sslstream = try!(SslStream::connect(context, connection));
+			let ssl = try!(Ssl::new(context));
+			let sslstream = ssl.connect(connection).unwrap();
```
Comment (nox) [inline src/client/mod.rs]: Pretty sure this should be in a `try!` expression.

File: src/header/accept.rs [hunk 3]
```diff
@@ -54,7 +54,7 @@ impl WebSocketAccept {
 		let mut concat_key = String::with_capacity(serialized.len() + 36);
 		concat_key.push_str(&serialized[..]);
 		concat_key.push_str(MAGIC_GUID);
-		let output = hash(hash::Type::SHA1, concat_key.as_bytes());
+		let output = hash(hash::MessageDigest::sha1(), concat_key.as_bytes()).unwrap();
```
Comment (nox) [inline src/header/accept.rs]: Why is this fallible now?

File: src/server/mod.rs [hunk 4]
```diff
@@ -60,7 +59,7 @@ pub mod response;
 ///let _ = context.set_private_key_file(&(Path::new("key.pem")), X509FileType::PEM);
 ///let server = Server::bind_secure("127.0.0.1:1234", &context).unwrap();
 ///
-///for connection in server {
+///for connection in server {		//FIXME
```
Comment (nox) [inline src/server/mod.rs]: What is there to fix here?

File: src/stream.rs [hunk 5]
```diff
@@ -80,7 +80,7 @@ impl WebSocketStream {
 	pub fn try_clone(&self) -> io::Result<WebSocketStream> {
 		Ok(match *self {
 			WebSocketStream::Tcp(ref inner) => WebSocketStream::Tcp(try!(inner.try_clone())),
-			WebSocketStream::Ssl(ref inner) => WebSocketStream::Ssl(try!(inner.try_clone())),
+			WebSocketStream::Ssl(_) => panic!("Cannot clone ssl stream"),
```
Comment (nox) [inline src/stream.rs:83]: That seems wrong.

**Review by nox — CHANGES_REQUESTED** [review]
Great!

File: src/stream.rs (hunk 5 above)
Comment (avadacatavra) [reply src/stream.rs:83]: in SslStream the try_clone(&self) function is deprecated in newer versions and will be removed

File: src/header/accept.rs (hunk 3 above)
Comment (avadacatavra) [reply src/header/accept.rs]: hash now outputs a result with an ssl errorstack. hashing can fail at a number of steps where openssl calls into c. if there's an error, we're just going to want to fail.

**Comment by avadacatavra** [comment]
I'm closing this in favor of my new PR #4 ...I did weird things with some configs and got all messed up.

**Pull request closed without merging**